tests/test_pyrflx.py

```diff
@@ -625,9 +630,9 @@ def test_ethernet_parse_binary(self) -> None:
             b"\x1e\x1f\x20\x21\x22\x23\x24\x25\x26\x27\x28\x29\x2a\x2b\x2c\x2d"
             b"\x2e\x2f\x30\x31\x32\x33\x34\x35\x36\x37"
         )
-        self.frame.parse(test_bytes)
-        self.assertTrue(self.frame.valid_message)
-        self.assertEqual(self.frame.bytestring, test_bytes)
+        self.frame_without_refinement.parse(test_bytes)
+        self.assertTrue(self.frame_without_refinement.valid_message)
+        self.assertEqual(self.frame_without_refinement.bytestring, test_bytes)
```

Comment:
Why do you test this with a version of `frame` without refinements? Would the result be different with `frame`?
- If yes, add the same test with `frame` to test the difference in behavior.
- If no, just use `frame`.

Reply:
I removed frame_without_refinements. Now, because the refinement is applied, i had to add support for the Always_Valid aspect in EnumValue. Subsequently the tests involving ethernet_vlan_tag.raw failed because of a wrong Total_Length value in the nested IPv4 packet. I changed the Total_Length value from x0014 to x002f and this fixed my tests in test_pyrflx but now some SPARK-Tests are failing.

Reply:
Please fix the failing SPARK tests. You only have to replace 20 by 47 in the affected tests (`tests/rflx-ethernet-tests.adb`).

Reply:
Fixed by bfda76d